rust_src/src/editfns.rs

```diff
@@ -799,4 +800,58 @@ pub fn char_equal(c1: LispObject, c2: LispObject) -> bool {
     unsafe { downcase(c1 as c_int) == downcase(c2 as c_int) }
 }
 
+/// Insert before point a substring of the contents of BUFFER.
+/// BUFFER may be a buffer or a buffer name.
+/// Arguments START and END are character positions specifying the substring.
+/// They default to the values of (point-min) and (point-max) in BUFFER.
+///
+/// Point and before-insertion markers move forward to end up after the
+/// inserted text.
+/// Any other markers at the point of insertion remain before the text.
+///
+/// If the current buffer is multibyte and BUFFER is unibyte, or vice
+/// versa, strings are converted from unibyte to multibyte or vice versa
+/// using `string-make-multibyte' or `string-make-unibyte', which see.
+#[lisp_fn(min = "1")]
+pub fn insert_buffer_substring(buffer: LispObject, beg: LispObject, end: LispObject) -> LispObject {
```

Comment:
So asking for `buffer: LispBuffer` is not enough?

Reply:
No, it also accepts a string containing a buffer name. We could make a type that would stand in for either one, though. Some kind of enum, perhaps, with it's own From impl.

Reply:
That sounds useful as this is a recurring pattern.

Reply:
Can we open an issue for it and merge this PR ? I assume you want the same for processes ?

Reply:
Oh yea, it doens't actually need to hold up this pr.

Reply:
Agreed, that was not my intent.

Reply:
I have a draft in the works BTW, once I have it working I will put it up for comment.